patroni/dcs/consul.py

```diff
@@ -324,12 +346,52 @@ def touch_member(self, data, ttl=None, permanent=False):
         try:
             args = {} if permanent else {'acquire': self._session}
             self._client.kv.put(self.member_path, json.dumps(data, separators=(',', ':')), **args)
+            if self._register_service:
+                self.update_service(self._my_member_data, data)
             self._my_member_data = data
             return True
         except Exception:
             logger.exception('touch_member')
         return False
 
+    @catch_consul_errors
+    def register_service(self, service_name, **kwargs):
+        logger.info('Register service {}, params {}'.format(service_name, kwargs))
+        return self.retry(self._client.agent.service.register, service_name, **kwargs)
```

Comment:
if we would fail registration silently then we will never retry to register again before one of `['role', 'api_url', 'conn_url']` is changed. Otherwise is it ok to fail entire `touch_member` if registration is failed?